internal/core/metadata/v2/controller/http/deviceservice.go

```diff
@@ -122,3 +122,51 @@ func (dc *DeviceServiceController) GetDeviceServiceByName(w http.ResponseWriter,
 	utils.WriteHttpHeader(w, ctx, statusCode)
 	pkg.Encode(response, w, lc)
 }
+
+func (dc *DeviceServiceController) GetAllDeviceServices(w http.ResponseWriter, r *http.Request) {
+	lc := container.LoggingClientFrom(dc.dic.Get)
+	ctx := r.Context()
+	correlationId := correlation.FromContext(ctx)
+
+	var offset = 0
+	var limit = 20
+	var labels = []string{}
+	var response interface{}
+	var statusCode int
+
+	// parse URL query string
+	offset, err := utils.ParseQueryStringToInt(r, QueryStringOffset)
+	if err != nil {
+		lc.Error(err.Error(), clients.CorrelationHeader, correlationId)
+		lc.Debug(err.DebugMessages(), clients.CorrelationHeader, correlationId)
+		response = commonDTO.NewBaseResponse("", err.Message(), err.Code())
+		statusCode = err.Code()
+	}
+
+	limit, err = utils.ParseQueryStringToInt(r, QueryStringLimit)
```

Comment:
Based on the `ParseQueryStringToInt` implementation, if `limit` is not provided this will result in it getting set to 0. Probably should pass in the default value to use when query parameter doesn't exist.

Reply:
Good catch!  I refactored this func with default value as part of func parameters.